db.go

```diff
@@ -489,8 +491,6 @@ func (db *DB) checkValue(chunk []byte) ([]byte, error) {
 	record := decodeLogRecord(chunk)
 	now := time.Now().UnixNano()
 	if record.Type == LogRecordDeleted || record.IsExpired(now) {
-		// delete from index if the key is expired.
-		db.index.Delete(record.Key)
```

Comment:
Deadlock? But we need this to skip the expired data while iterating the entire db.

Reply:
index.Ascend/Descend/… hold index.Rlock, and index.Delete need index.Lock, so it will deadlock.
When traversing, we can't simply delete key. db.index use btree. If we delete the key, it's possible to rebalance, so the traversal data will not be accurate.
If it is necessary to delete Expired keys, it is more appropriate to use db.DeleteExpiredKeys.